### ISSUE TITLE
Call to undefined method during (and only during) the update from 1.5.0 to 1.7.0

### ISSUE THREAD

**Issue by AndreasA** [bamarni/composer-bin-plugin#125]
It is related to https://github.com/bamarni/composer-bin-plugin/issues/122

However, the issue does not happen always, it just happens during the update.

Using PHP 8.1 and composer 2.3.7

So it works fine in these scnearios:
- After the update all `composer bin` plugin commands and composer commands work fine.
- After the update, removing vendor/bamarni/composer-bin-plugin and running composer install.
- After the update, removing vendor completely and running composer install works fine as well.

What fails:

- Updating from 1.5.0 to 1.7.0 produces an error message, but the composer.lock seems to have been updated, like in the issue https://github.com/bamarni/composer-bin-plugin/issues/122:
   ````
   In PluginManager.php : eval()'d code line:
                                                                        
  [Error]                                                               
  Call to undefined method Bamarni\Composer\Bin\Config::fromComposer()  
  ```
  (removed code lines as I am currently on a different system, just copied the general message from the referenced bug as it was the same).

As everything works fine after the update, it isn't that huge an issue but should probably still be looked into.

**Comment by theofidry** [comment]
Closing as a duplicate of #124 which should be fixed by #132 :)

**Comment by AndreasA** [comment]
@theofidry 

I was using composer 2.3.7 when I updated from 1.5.0 to 1.7.0, so this should be unrelated? And as mentioned it only occured during the update, afterwards it was working fine.

Also the error message is different. Here it seems that `fromComposer` is missing, in yours it is `requireComposer` but maybe the are related.
